dspace-server-webapp/src/main/java/org/dspace/app/rest/OrcidRestController.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.net.URI;
+import java.sql.SQLException;
+import java.util.Optional;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.dspace.app.orcid.client.OrcidClient;
+import org.dspace.app.orcid.model.OrcidTokenResponseDTO;
+import org.dspace.app.orcid.service.OrcidSynchronizationService;
+import org.dspace.app.profile.ResearcherProfile;
+import org.dspace.app.rest.model.RestModel;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.services.ConfigurationService;
+import org.dspace.util.UUIDUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Rest controller that store ORCID infos and handles redirect.
+ *
+ * @author Luca Giamminonni (luca.giamminonni at 4science.it)
+ */
+@RequestMapping(value = "/api/" + RestModel.EPERSON + "/orcid")
+@RestController
+public class OrcidRestController {
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Autowired
+    private OrcidClient orcidClient;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private OrcidSynchronizationService orcidSynchronizationService;
+
+    @GetMapping(value = "/{itemId}")
+    public void linkProfileFromCode(HttpServletRequest request, HttpServletResponse response,
```

Comment:
I can't find this in the REST contract

Reply:
Every REST endpoint should be in the contract, but you can explain in the contract why it's not meant for common use

Reply:
I've tested this new PATCH request approach today (using the frontend PR https://github.com/DSpace/dspace-angular/pull/1618).  It works well & it resolves my feedback with regards to this endpoint.  The `PATCH` seems appropriate here & it means this entire Controller is removed.   I see this new `PATCH` option is also appropriately added to the REST Contract (with the old GET removed): https://github.com/DSpace/RestContract/pull/190 Thanks @abollini and @LucaGiamminonni